src/rulesets/oas/functions/oasOpParams.ts

```diff
@@ -1,138 +1,69 @@
-              inBoth[paramIndex] = params[paramIndex];
-              inBoth[compareIndex] = params[compareIndex];
-            }
+  if (!Array.isArray(params)) return;
 
-            // Operation must have only one `in:body` parameter.
-            if (
-              // skip param that already tested true
-              !inBody[paramIndex] &&
-              params[paramIndex].in === 'body' &&
-              params[compareIndex].in === 'body'
-            ) {
-              inBody[paramIndex] = params[paramIndex];
-              inBody[compareIndex] = params[compareIndex];
-              //
-            }
-          }
-        }
+  if (params.length < 2) return;
 
-        if (Object.keys(nonUnique).length > 0) {
-          for (const group in nonUnique) {
-            if (!nonUnique[group]) continue;
-
-            const firstMatch = Object.keys(nonUnique[group])[0];
-            const inVal = nonUnique[group][firstMatch].in;
-            const nameVal = nonUnique[group][firstMatch].name;
-
-            let message = `Operations must have unique \`name\` + \`in\` parameters.\nRepeats of \`in:${inVal}\` + \`name:${nameVal}\`\n\nParameters found at:\n`;
-
-            for (const index in nonUnique[group]) {
-              if (!inBody[index]) {
-                message += `[ \`paths\`, \`${path}\`, \`${operation}\`, \`${index}\` ]\n`;
-              }
-            }
-
-            results.push({
-              message,
-              path: ['paths', path, operation],
-            });
-          }
-        }
-
-        if (Object.keys(inBoth).length > 1) {
-          let message = 'Operation cannot have both `in:body` and `in:formData` parameters.\n\nParameters found at:\n';
+  const results: IFunctionResult[] = [];
 
-          for (const index in inBoth) {
-            if (!inBody[index]) {
-              message += `[ \`paths\`, \`${path}\`, \`${operation}\`, \`${index}\` ]\n`;
-            }
-          }
+  const count = {
+    body: 0,
+    formData: 0,
+  };
+  const list: string[] = [];
+  let hasDuplicateParams = false;
+
+  for (const param of params) {
+    if (param === null || typeof param !== 'object') continue;
+
+    // skip params that are refs
+    if ('$ref' in param) continue;
+
+    // Operations must have unique `name` + `in` parameters.
+    if (!hasDuplicateParams) {
+      const fingerprint = computeFingerprint(param);
+      if (list.includes(fingerprint)) {
+        hasDuplicateParams = true;
+      } else {
+        list.push(fingerprint);
+      }
+    }
 
-          results.push({
-            message,
-            path: ['paths', path, operation],
-          });
-        }
+    if (param.in in count) {
+      count[param.in]++;
+    }
+  }
 
-        if (Object.keys(inBody).length > 1) {
-          let message = 'Operation has multiple instances of the `in:body` parameter.\n\nParameters found at:\n';
+  if (hasDuplicateParams) {
+    results.push({
+      message: 'Operation must have unique `name` + `in` parameters',
```

Comment:
@P0lip Couldn't we point to the first duplicated parameter and issue an error message like

```
A parameter in this operation already exposes the same combination of `name` and `in` values.
```

@rossmcdonald  Thoughts?

Reply:
We'd need to list all duplicate parameters in this case, wouldn't we?
I thought of it, and both ideas sound good to me.
I'll let @philsturgeon and @rossmcdonald decide.

Reply:
>We'd need to list all duplicate parameters in this case, wouldn't we?

@P0lip You're right. This is more or less how `oasPathParams` work.  Basically when 3 entries are considered as duplicates, the two last ones are returned as "duplicated" results.

Reply:
Let's do it in the same way then to be consistent.